src/arviz_base/transform.py

```diff
@@ -0,0 +1,100 @@
+"""Transform helper."""
+
+from collections.abc import Callable
+from copy import deepcopy
+
+import xarray as xr
+
+from arviz_base.utils import _var_names
+
+
+def get_unconstrained_samples(
+    idata,
+    var_names=None,
+    filter_vars=None,
+    transform_funcs: dict[str, Callable[[xr.DataArray], xr.DataArray]] | None = None,
+    group="posterior",
+    return_dataset=False,
+):
+    """Transform helper.
+
+    It transform the variables of a DataTree in a given group applying the functions
+    in the dictionary `transform_funcs`.
+    With the default values, it would combine the samples in posterior and unconstrained posterior
+    to have all the samples in the unconstrained space.
+
+    Parameters
+    ----------
+    idata : DataTree-like
+        DataTree from which to extract the data.
+    var_names : str or list of str, optional
```

Comment:
I think if we use the following we'll get a better type hint (I know we probably do have bad type hints in multiple places, but I think at least list and tuples are valid), but you know more about type hints so let me know if you think it should not be incorporated:

```suggestion
    var_names : str or sequence of str, optional
```